src/reader.rs

```diff
@@ -114,26 +112,12 @@ pub trait Reader: Debug + Clone + Read {
 
     /// Read an unsigned LEB128 encoded integer.
     fn read_uleb128(&mut self) -> Result<u64> {
-        match leb128::read::unsigned(self) {
-            Ok(val) => Ok(val),
-            Err(leb128::read::Error::IoError(ref e)) if e.kind() ==
-                                                        io::ErrorKind::UnexpectedEof => {
-                Err(Error::UnexpectedEof)
-            }
-            Err(_) => Err(Error::BadUnsignedLeb128),
-        }
+        leb128::read::unsigned(self)
     }
 
     /// Read a signed LEB128 encoded integer.
     fn read_sleb128(&mut self) -> Result<i64> {
-        match leb128::read::signed(self) {
-            Ok(val) => Ok(val),
-            Err(leb128::read::Error::IoError(ref e)) if e.kind() ==
-                                                        io::ErrorKind::UnexpectedEof => {
-                Err(Error::UnexpectedEof)
-            }
-            Err(_) => Err(Error::BadSignedLeb128),
-        }
+        leb128::read::signed(self)
```

Comment:
Might make sense for these functions to either go away or be `#[inline]` too now (although I suspect rustc/llvm is doing the expected inlining anyways).